tests/appsec/test_automated_login_events.py

```diff
@@ -1347,3 +1220,538 @@ def validate_iden(meta):
+
+    def setup_login_wrong_password_failure_local(self):
+        self.r_wrong_user_failure = weblog.post("/login?auth=local", data=login_data(context, USER, "12345"))
+
+    def test_login_wrong_password_failure_local(self):
+        assert self.r_wrong_user_failure.status_code == 401
+        for _, trace, span in interfaces.library.get_spans(request=self.r_wrong_user_failure):
+            assert_priority(span, trace)
+            meta = span.get("meta", {})
+
+            # mandatory
+            assert meta["appsec.events.users.login.failure.usr.login"] == USER
+            assert meta["_dd.appsec.usr.login"] == USER
+            assert meta["_dd.appsec.events.users.login.failure.auto.mode"] == "identification"
+            assert meta["appsec.events.users.login.failure.track"] == "true"
+
+            # optional (to review for each library)
+            assert meta["appsec.events.users.login.failure.usr.exists"] == "false"
+            assert meta["appsec.events.users.login.failure.usr.id"] == "social-security-id"
+            assert meta["_dd.appsec.usr.id"] == "social-security-id"
+
+    def setup_login_wrong_password_failure_basic(self):
+        self.r_wrong_user_failure = weblog.get(
+            "/login?auth=basic", headers={"Authorization": BASIC_AUTH_INVALID_PASSWORD_HEADER}
+        )
+
+    @missing_feature(context.library == "php", reason="Basic auth not implemented")
+    def test_login_wrong_password_failure_basic(self):
+        assert self.r_wrong_user_failure.status_code == 401
+        for _, trace, span in interfaces.library.get_spans(request=self.r_wrong_user_failure):
+            assert_priority(span, trace)
+            meta = span.get("meta", {})
+
+            # mandatory
+            assert meta["appsec.events.users.login.failure.usr.login"] == USER
+            assert meta["_dd.appsec.usr.login"] == USER
+            assert meta["_dd.appsec.events.users.login.failure.auto.mode"] == "identification"
+            assert meta["appsec.events.users.login.failure.track"] == "true"
+
+            # optional (to review for each library)
+            assert meta["appsec.events.users.login.failure.usr.exists"] == "false"
+            assert meta["appsec.events.users.login.failure.usr.id"] == "social-security-id"
+            assert meta["_dd.appsec.usr.id"] == "social-security-id"
+
+    def setup_login_sdk_success_local(self):
+        self.r_sdk_success = weblog.post(
+            "/login?auth=local&sdk_event=success&sdk_user=sdkUser", data=login_data(context, USER, PASSWORD),
+        )
+
+    def test_login_sdk_success_local(self):
+        assert self.r_sdk_success.status_code == 200
+        for _, trace, span in interfaces.library.get_spans(request=self.r_sdk_success):
+            assert_priority(span, trace)
+            meta = span.get("meta", {})
+
+            # mandatory
+            assert meta["appsec.events.users.login.success.usr.login"] == "sdkUser"
```

Comment:
I think this is wrong. The SDK isn't supposed to set the usr.login to "sdkUser", it's supposed to set the usr.id right ?

Reply:
Let's sync on this during today's meeting, as we discuss during the last one to update the SDK in order to start using the login instead of the user id.

Reply:
We decided to implement this change now :+1: